recipes/hatchling/meta.yaml

```diff
@@ -0,0 +1,50 @@
+{% set name = "hatchling" %}
+{% set version = "0.15.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/hatchling-{{ version }}.tar.gz
+  sha256: 33816429b78d0f069717ccce53a2ec722c79412c86928caa2e01a93b4eaf31a5
+
+build:
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+  entry_points:
+    - hatchling = hatchling.cli:hatchling
+
+requirements:
+  host:
+    - pip
+    - python
+  run:
+    - editables >=0.2  # [py>3],==0.*
+    - importlib-metadata  # [py<38]
+    - packaging >=20.9  # [py<3],==20.*
+    - pathspec >=0.9,==0.*
+    - pluggy >=0.13  # [py<3],==0.*
+    - python
+    - toml >=0.10.2  # [py<3],==0.10.*
+    - tomli >=2.0.0  # [py>3],==2.0.*
```

Comment:
```suggestion
    - editables >=0.2
    - importlib-metadata
    - packaging >=20.9
    - pathspec >=0.9
    - pluggy >=0.13
    - python
    - tomli >=2.0
```

Reply:
Conda-forge doesn't support Python 2 anymore, and you can't set Python requirements on a pure Python package (which this should be, `noarch: python`), and [no upper caps](https://iscinumpy.dev/post/bound-version-constraints/), please. They can be injected later if something breaks.